Tool/FixtureGenerator.php

```diff
@@ -326,8 +331,42 @@ public function generateFixtureItemStub($item)
 
         $code = "";
         $reflexion = new \ReflectionClass($item);
+
+        /** @var array $constructorArguments */
+        $constructorArguments = [];
+
+        /** @var \ReflectionMethod $constructor */
+        if ($constructor = $reflexion->getConstructor()) {
+            /** @var AnnotationReader $reader */
+            $reader = new AnnotationReader();
+            /** @var ConstructorArguments $constructorArguments */
+            $constructorArgumentsAnnotation = $reader->getClassAnnotation(
+                $reflexion,
+                'Webonaute\DoctrineFixturesGeneratorBundle\Annotation\ConstructorArguments'
+            );
+
+            if ($constructorArgumentsAnnotation) {
+                foreach ($constructorArgumentsAnnotation->value as $k => $ca) {
+                    if (isset($ca['value'])) {
+                        $caValue = $ca['value'];
+                    } elseif (isset($ca['php'])) {
+                        eval('$caValue = ' . $ca['php'] . ';');
+                    } else {
+                        throw new \Exception(sprintf('Cannot determine constructor argument. [$s]', json_encode($ca)));
+                    }
+
+                    $constructorArguments[$k] = $caValue;
+                }
+            }
+        }
+
         $properties = $this->getRecursiveProperties($reflexion);
-        $newInstance = $reflexion->newInstance();
+
+        if (count($constructorArguments) > 0) {
```

Comment:
this need to go in its own method. getNewInstanceFromReflexion($reflexion, $constructorArguments).